src/ORM/Association/HasMany.php

```diff
@@ -117,6 +165,10 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
                 $targetEntity->set($properties, ['guard' => false]);
             }
 
+            if ($this->_saveStrategy === self::SAVE_REPLACE) {
+                $targetEntity->isNew(true);
```

Comment:
They were previously deleted (unlinked), because of the replace saveStrategy.
If they weren't set as new, they wouldn't be re-added.